src/ethereum-json-rpc-client/src/lib.rs

```diff
@@ -252,8 +262,20 @@ impl<C: Client> EthJsonRcpClient<C> {
 }
 
 pub trait Client: Clone + Send + Sync {
-    fn send_rpc_request(
+    /// Send RPC request.
+    ///
+    fn send_request(
         &self,
-        request: Request,
+        request: ClientRequest,
     ) -> Pin<Box<dyn Future<Output = anyhow::Result<Response>> + Send>>;
 }
+
+/// An enum representing the different clients which are supported
+/// for sending request.
+///
+/// It can either use HttpOutCalls or Rpc Request
+#[derive(Debug)]
+pub enum ClientRequest {
+    RpcRequest(Request),
+    HttpOutCall(HttpOutCallArgs),
+}
```

Comment:
I think we should use the same `jsonrpc-core::Request` for all implementations, because we are describing an abstract JSON-RPC client here.

Additional data for HttpOutcall (URL, HTTP method, max bytes, cycles calculation) could be implemented as a part of `HttpOutcallClient`:
- We can store an URL as a field `HttpOutcallClient::url`.
- We can select http method, max bytes, cycles cost, using data from `jsonrpc-core::Request`. For example, we can allow client to add information about methods he wants to use: 
```rust
pub struct HttpOutcallClient<T: MethodMetadataProvider> {
    url: String,
    method_metadata: T
}

trait MethodMetadataProvider {
    fn get_metadata(request: &jsonrpc_core::Request) -> MethodMetadata;
}

struct MethodMetadata {
    http_method: HttpMethod,
    max_response_bytes: u64,
    cycles_cost: u64,
}
```
So, for each json-rpc call, we can internally select parameters.

This approach will make usage of HttpOutcallClient much simpler in generic code.